app/models/reservation.rb

```diff
@@ -73,8 +73,12 @@ def confirmed?
     status == Status::CONFIRMED
   end
 
+  def cancelable?
+    (Status::CANCELABLE.include? status) && ((self.start.to_i-Time.now.to_i) > (self.nook.modifiable_before*3600))
+  end
+
   def modifiable?
-    Status::MODIFIABLE.include? status
+    (Status::MODIFIABLE.include? status) && ((self.start.to_i-Time.now.to_i) > (self.nook.modifiable_before*3600))
```

Comment:
@shlok007 Sorry for noticing this late - though you have updated this function, it was in fact used for admin. The 'My reservations' for patron view still give the option to modify.